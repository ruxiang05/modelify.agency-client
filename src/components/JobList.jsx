import React from 'react';
import JobCard from './JobCard';
import { getToken } from '../auth';
import api from '../api';
import PageHeader from './PageHeader';

class JobList extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      jobs: [],
    };
  }

  componentDidMount() {
    const token = getToken();
    api.jobs.getJobs(token).then((res) => {
      const { jobs } = res;
      if (jobs) this.setState({ jobs });
    });
  }

  render() {
    const { jobs } = this.state;
    return (
      <React.Fragment>
        <PageHeader title="Jobs" />
        <div className="page">
          {jobs.length ? (
            <ul>
              {jobs.map(job => (
                <li key={jobs.indexOf(job)}>
                  <JobCard job={job} />
                </li>
              ))}
            </ul>
          ) : (
            <div>No jobs yet</div>
          )}
        </div>
      </React.Fragment>
    );
  }
}

export default JobList;
