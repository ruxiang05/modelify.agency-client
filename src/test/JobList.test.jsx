/* Uses methods and/or components from react and enzyme */
import React from 'react';
import { shallow } from 'enzyme';
import JobList from '../components/JobList';

jest.mock('../api');

describe('JobList', () => {
  it('should render JobList', () => {
    const wrapper = shallow(<JobList />);
    expect(wrapper).toMatchSnapshot();
  });

  it('show a list of jobs', async () => {
    const wrapper = shallow(<JobList />);
    wrapper.instance().componentDidMount();
    await wrapper.update();
    expect(wrapper.state('jobs').length).toBe(1);
  });
});
