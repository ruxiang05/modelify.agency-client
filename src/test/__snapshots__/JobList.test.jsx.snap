// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobList should render JobList 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <JobList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <PageHeader
          backButton={false}
          title="Jobs"
        />,
        <div
          className="page"
        >
          <NoData>
            <p>
              You do not have jobs yet
            </p>
          </NoData>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "action": undefined,
          "actionAltText": undefined,
          "actionIcon": undefined,
          "backButton": false,
          "goBackAction": undefined,
          "history": undefined,
          "title": "Jobs",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NoData>
            <p>
              You do not have jobs yet
            </p>
          </NoData>,
          "className": "page",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <p>
              You do not have jobs yet
            </p>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "You do not have jobs yet",
            },
            "ref": null,
            "rendered": "You do not have jobs yet",
            "type": "p",
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <PageHeader
            backButton={false}
            title="Jobs"
          />,
          <div
            className="page"
          >
            <NoData>
              <p>
                You do not have jobs yet
              </p>
            </NoData>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "action": undefined,
            "actionAltText": undefined,
            "actionIcon": undefined,
            "backButton": false,
            "goBackAction": undefined,
            "history": undefined,
            "title": "Jobs",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NoData>
              <p>
                You do not have jobs yet
              </p>
            </NoData>,
            "className": "page",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <p>
                You do not have jobs yet
              </p>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "You do not have jobs yet",
              },
              "ref": null,
              "rendered": "You do not have jobs yet",
              "type": "p",
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
